### PR TITLE
[CVE-2022-23852] Prevent XML_GetBuffer signed integer overflow

### PR DESCRIPTION
**Pull request by hartwork**
CC @ferivoz @rillig

### CODE REVIEW

File: expat/tests/runtests.c [hunk 1]
```diff
@@ -3847,6 +3847,30 @@ START_TEST(test_get_buffer_2) {
 }
 END_TEST
 
+/* Test for signed integer overlow CVE-TODO */
```
Comment (rillig) [inline expat/tests/runtests.c:3850]: typo: "overlow"

**Comment by rillig** [comment]
Looks good to me as the code uses the same pattern as everywhere else to detect integer overflow.

File: expat/tests/runtests.c (hunk 1 above)
Comment (hartwork) [reply expat/tests/runtests.c]: Good catch, fixed, thanks! :+1:

**Comment by hartwork** [comment]
@rillig thanks for the review :+1: 

CVE request submitted to Mitre just now.

**Comment by hartwork** [comment]
Got CVE CVE-2022-23852

**Comment by ferivoz** [comment]
Looks good to me as well. Thank you for extending the test into its own test case and adjusting the signed overflow check!

**Comment by hartwork** [comment]
@ferivoz thanks for the review :+1: 

Merging now...

**Comment by iamkarlson** [comment]
Can anyone who submitted a CVE explain in more detail what is this package for and what might be affected by using it?

**Comment by hartwork** [comment]
@iamkarlson are you asking what libexpat is being used for in general? Have you read the project description? For an (incomplete) list of hardware and other Open Source software using libexpat please see https://libexpat.github.io/doc/users/ .

**Comment by DominikBauer1** [comment]
Hi, the CVSS value of 9.8 seems quite high for an integer overflow. Is there any malicious advantage that I don't see at the moment?

**Comment by hartwork** [comment]
@DominikBauer1 to my best knowledge it is unclear how much more than DoS can be gotten out of that overflow: The optimistic view is "probably just DoS" and the pessimistic one is "this could be code execution". That's why I wrote "Impact is denial of service or more" in the change log. I didn't suggest a score that high, but maybe the person who made that choice has information that I don't. Does that answer the question?

**Comment by iamkarlson** [comment]
@hartwork yes, that's what I was looking for. We're using docker images and I wanted to remove all unpatched dependencies from them.

**Pull request merged**